### PR TITLE
[Documentation] Add model serving example using BentoML

### PR DESCRIPTION
**Pull request by yubozhao**
Add model serving example using BentoML.

[BentoML](https://bentoml.org) is an open-source platform for high-performance ML model serving.

This example demos Kubeflow users another option to deploy their model to Kubeflow enabled Kubernetes cluster.

### CODE REVIEW

**Comment by sarahmaddox** [comment]
/ok-to-test

Preview: 
* https://deploy-preview-1927--competent-brattain-de2d6d.netlify.app/docs/components/serving/bentoml/
* https://deploy-preview-1927--competent-brattain-de2d6d.netlify.app/docs/components/serving/overview/

File: content/en/docs/components/serving/bentoml.md [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+title = "BentoML"
+description = "Model serving with BentoML"
+weight = 40
++++
+
+{{% stable-status %}}
+
+## Model serving with BentoML
+
+In this guide, we will demo serving a scikit-learn based iris classifier model with
```
Comment (sarahmaddox) [inline content/en/docs/components/serving/bentoml.md:11]: A technical writing hint: It's better to avoid the first person ("we") because the reader can become confused. It's not clear who "we" is in this context.

Suggestion:

```
This guide demonstrates how to serve a scikit-learn...
```

File: content/en/docs/components/serving/overview.md [hunk 2]
```diff
@@ -188,3 +188,20 @@ You can use NVIDIA Triton Inference Server as a
...
+[BentoML](https://bentoml.org) is an open-source platform for high-performance ML model
+serving. It makes building production API endpoint for your ML model easy and supports
+all major machine learning training frameworks, including Tensorflow, Keras, PyTorch,
+XGBoost, scikit-learn and etc.
+
+BentoML comes with a high-performance API model server with adaptive micro-batching
+support, which achieves the advantage of batch processing in online serving. It also
+provides model management and model deployment functionality, giving ML teams an
+end-to-end model serving workflow, with DevOps best practices baked in.
+
```
Comment (sarahmaddox) [inline content/en/docs/components/serving/overview.md:203]: Add a link here that points to the BentoML guide in Kubeflow:

```
* [BentoML guide for Kubeflow](/docs/components/serving/bentoml)
```

File: content/en/docs/components/serving/bentoml.md [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+curl -i \
+  --header "Content-Type: application/json" \
+  --request POST \
+  --data '[[5.1, 3.5, 1.4, 0.2]]' \
+  http://EXTERNAL_IP:NODE_PORT/predict
+```
+
+### Monitor metrics with Prometheus
+
+#### Perquisites
```
Comment (sarahmaddox) [inline content/en/docs/components/serving/bentoml.md:199]: Fix typo `Prerequisites`.

File: content/en/docs/components/serving/bentoml.md [hunk 4]
```diff
@@ -0,0 +1,258 @@
++++
+title = "BentoML"
+description = "Model serving with BentoML"
+weight = 40
++++
+
+{{% stable-status %}}
+
+## Model serving with BentoML
```
Comment (sarahmaddox) [inline content/en/docs/components/serving/bentoml.md:9]: Recommendation: Remove this heading, as it's basically a repeat of the page title/description. Then change all the other headings from level 3 to level 2.

File: content/en/docs/components/serving/bentoml.md [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+all major machine learning training frameworks, including Tensorflow, Keras, PyTorch,
+XGBoost, scikit-learn and etc.
+
+BentoML comes with a high-performance API model server with adaptive micro-batching
+support, which achieves the advantage of batch processing in online serving. It also
+provides model management and model deployment functionality, giving ML teams an
+end-to-end model serving workflow, with DevOps best practices baked in.
+
+### Prerequisites
+
```
Comment (sarahmaddox) [inline content/en/docs/components/serving/bentoml.md:27]: Recommendation: Add an introductory sentence here, to introduce the list and to help people who may not understand the word "prerequisites":

```
Before starting this tutorial, make sure you have the following:
```

File: content/en/docs/components/serving/bentoml.md [hunk 6]
```diff
@@ -0,0 +1,258 @@
...
+1. a Kubernetes cluster and `kubectl` installed on your local machine.
```
Comment (sarahmaddox) [inline content/en/docs/components/serving/bentoml.md:28]: Change the numbered list to a bulleted list. (Tech writing style is to use a numbered list when the order is important, such as in a sequence of steps, and a bulleted list in other situations.)

File: content/en/docs/components/serving/bentoml.md [hunk 7]
```diff
@@ -0,0 +1,258 @@
...
+description = "Model serving with BentoML"
+weight = 40
++++
+
+{{% stable-status %}}
+
+## Model serving with BentoML
+
+In this guide, we will demo serving a scikit-learn based iris classifier model with
+BentoML on a Kubernetes cluster. The same deployment steps is also applicable for models
```
Comment (sarahmaddox) [inline content/en/docs/components/serving/bentoml.md:12]: Change "is" to "are".

File: content/en/docs/components/serving/bentoml.md [hunk 8]
```diff
@@ -0,0 +1,258 @@
...
+    * `kubectl` install instruction: https://kubernetes.io/docs/tasks/tools/install-kubectl/
+2. Docker and Docker Hub installed and configured in your local machine.
+    * Docker install instruction: https://docs.docker.com/get-docker/
+3. Python 3.6 or above and required PyPi packages: `bentoml`, `scikit-learn`
+    * ```pip install bentoml scikit-learn```
+
+### Build an iris classifier model server with BentoML
+
+The following code defines a BentoML prediction service that requires a `scikit-learn` model, and
+asks BentoML to figure out the required PyPI packages automatically. It also defined an
```
Comment (sarahmaddox) [inline content/en/docs/components/serving/bentoml.md:38]: s/defined/defines

File: content/en/docs/components/serving/bentoml.md [hunk 9]
```diff
@@ -0,0 +1,258 @@
...
+@env(auto_pip_dependencies=True)
+@artifacts([SklearnModelArtifact('model')])
+class IrisClassifier(BentoService):
+
+    @api(DataframeHandler)
+    def predict(self, df):
+        return self.artifacts.model.predict(df)
+```
+
+The following code trains a classifier model and serve it with the IrisClassifier
```
Comment (sarahmaddox) [inline content/en/docs/components/serving/bentoml.md:57]: s/serve/serves

File: content/en/docs/components/serving/bentoml.md [hunk 10]
```diff
@@ -0,0 +1,258 @@
...
+  --header "Content-Type: application/json" \
+  --request POST \
+  --data '[[5.1, 3.5, 1.4, 0.2]]' \
+  localhost:5000/predict
+```
+
+BentoML provides a convenient way of containerizing the model API server with Docker. To
+create a docker container image for the sample model above:
+
+Find the file directory of the SavedBundle with `bentoml get` command, which is
```
Comment (sarahmaddox) [inline content/en/docs/components/serving/bentoml.md:114]: Suggestion: Change this paragraph to a list. The introductory sentence above the paragraph implies that there'll be a list, because it ends with a colon. Also, a list is probably easier to read in this case.

File: content/en/docs/components/serving/bentoml.md [hunk 11]
```diff
@@ -0,0 +1,258 @@
...
+saved_path=$(bentoml get IrisClassifier:latest -q | jq -r ".uri.uri")
+
+docker build -t {docker_username}/iris-classifier $saved_path
+docker push {docker_username}/iris-classifier
+```
+
+### Deploy model server to Kubernetes
+
+The following is an example YAML file for specifying the resources required to run and
+expose a BentoML model server in a Kubernetes cluster. Replacing `{docker_username}`
```
Comment (sarahmaddox) [inline content/en/docs/components/serving/bentoml.md:129]: s/Replacing/Replace

File: content/en/docs/components/serving/bentoml.md [hunk 12]
```diff
@@ -0,0 +1,258 @@
...
+    spec:
+      containers:
+      - image: {docker_username}/iris-classifier
+        imagePullPolicy: IfNotPresent
+        name: iris-classifier
+        ports:
+        - containerPort: 5000
+```
+
+Use `kubectl` CLI to deploy the model API server to the kubernetes cluster
```
Comment (sarahmaddox) [inline content/en/docs/components/serving/bentoml.md:173]: s/kubernetes/Kubernetes

**Comment by yubozhao** [comment]
@sarahmaddox Thanks for reviewing and making those suggestions.  I will update the PR base on your comments.

File: content/en/docs/components/serving/bentoml.md (hunk 1 above)
Comment (yubozhao) [reply content/en/docs/components/serving/bentoml.md:11]: Thanks for the tips. I am still very new at technical writing, love to get more feedback on how I can improve it

**Comment by sarahmaddox** [comment]
Thanks!
/lgtm
/approve
